File: JavaPortfolio/SnakeFillGH.java
/*Tanner FitzGerald
 *8/27/24
 *Personal practice project
 *
 * This Progam will simulate the videogame "snake" somewhat.
 * It contains a function that will take a given length of a game board
 * side and return the amount of times the snake can eat before it runs
 * out of area
 * 
 * The snake starts at 1, and doubles with every apple eaten
 * 
 * This was one of my first Java programs, and so it has many glaring "learing moments" as I
 * was figuring out how to transition from python.
 * 
 */

import java.util.Scanner;

 public class SnakeFillGH{

    public static void main(String[] args) {

        //prompt user for side length, convert to integer
        Scanner myScanner = new Scanner(System.in);
        System.out.print("Enter Length of side: ");
        String sideLength = myScanner.nextLine();
        Integer intSideLength = Integer.valueOf(sideLength);

        //establish grid area, snake length, apple counter
        Integer area = intSideLength * intSideLength;
        Integer snake = 1;
        Integer apples = 0;

        //eat the apples
        while (snake < area){
            snake *= 2;
            apples += 1;
        }
        System.out.println(apples - 1);
        myScanner.close();
    }
 }

File: pythonPortfolio/snakeFillGH.py
#Tanner FitzGerald
#8/27/24
#Personal Practice Project
'''

This program will simulate a "snake" videogame somewhat, and contains a function
that will take a variable length of a game board size, and retrun the number of
times the snake can eat(and double its length) before it runs out of area. 

The snake starts at length 1.

'''

def snakefill(side_length):
    area = side_length * side_length
    snake = 1
    apples = 0
    while snake < area:
        snake *= 2
        apples += 1
    return str(apples - 1)

print(snakefill(int(input("Enter the length of one side of the game board "))))